Traceback is not set on future result value

This will result in the exception being displayed only containing the exception value and not the traceback, e.g.:
```
../aiida_core/aiida/engine/launch.py:36: in run
    return runner.run(process, *args, **inputs)
../aiida_core/aiida/engine/runners.py:223: in run
    result, _ = self._run(process, *args, **inputs)
../aiida_core/aiida/engine/runners.py:211: in _run
    process.execute()
../../../env/aiida/lib/python2.7/site-packages/plumpy/processes.py:88: in func_wrapper
    return func(self, *args, **kwargs)
../../../env/aiida/lib/python2.7/site-packages/plumpy/processes.py:1063: in execute
    return self.future().result()
../../../env/aiida/lib/python2.7/site-packages/plumpy/futures.py:34: in result
    return super(Future, self).result(timeout)
../../../env/aiida/lib/python2.7/site-packages/tornado/concurrent.py:238: in result
    raise_exc_info(self._exc_info)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

exc_info = (<class 'aiida.common.exceptions.NotExistent'>, NotExistent('no neighbor with the label output_structure found',), None)

>   ???
E   NotExistent: no neighbor with the label output_structure found
```
Without the original stack trace, this becomes impossible to debug.

--- a/plumpy/processes.py
+++ b/plumpy/processes.py
@@ -738,7 +738,9 @@ class Process(StateMachine, persistence.Savable):
 
     @super_check
     def on_except(self, exc_info):
-        self.future().set_exception(exc_info[1])
+        exception = exc_info[1]
+        exception.__traceback__ = exc_info[2]
+        self.future().set_exception(exception)
 
     @super_check
     def on_excepted(self):
@@ -1090,8 +1092,7 @@ class Process(StateMachine, persistence.Savable):
                 raise
             except Exception:  # pylint: disable=broad-except
                 # Overwrite the next state to go to excepted directly
-                exc_info = sys.exc_info()
-                next_state = self.create_state(process_states.ProcessState.EXCEPTED, exc_info[1], exc_info[2])
+                next_state = self.create_state(process_states.ProcessState.EXCEPTED, *sys.exc_info()[1:])
                 self._set_interrupt_action(None)
 
             if self._interrupt_action:

